File: matlab/tests/system/test_toolset_aout.m
%% test_toolset_aout.m - Run AOUT toolset on analog ADC data
close all; clc; clear; warning("off")

%% Configuration
verbose = 0;
inputDir = "dataset";
outputDir = "test_output";
filesList = {"sinewave_amplitude_noise_0P001.csv"};
filesList = autoSearchFiles(filesList, inputDir, 'sinewave_*.csv', 'batch_sinewave_*.csv');
if ~isfolder(outputDir), mkdir(outputDir); end

%% ADC Configuration
Resolution = 11;  % ADC resolution in bits

%% Test Loop
for k = 1:length(filesList)
    currentFilename = filesList{k};
    dataFilePath = fullfile(inputDir, currentFilename);
    fprintf('[%s] [%d/%d] [%s]\n', mfilename, k, length(filesList), currentFilename);
    [~, datasetName, ~] = fileparts(currentFilename);
    subFolder = fullfile(outputDir, datasetName, mfilename);

    aout_data = readmatrix(dataFilePath);
    status = toolset_aout(aout_data, subFolder, 'Visible', verbose, 'Resolution', Resolution);
end

File: python/tests/batch/test_toolset_aout.py
"""Test AOUT toolset on analog ADC data."""

import numpy as np
from pathlib import Path
import sys
import os

# Add project root to sys.path
_project_root = Path(__file__).parent.parent.parent.parent
if str(_project_root) not in sys.path:
    sys.path.insert(0, str(_project_root))

from python.src.adctoolbox.toolset_aout import toolset_aout

# Configuration
verbose = False
input_dir = Path('dataset') / 'aout'
output_dir = Path('test_output')
resolution = 11

# Get list of files
files_list = sorted(input_dir.glob('sinewave_*.csv'))
if not files_list:
    print('No sinewave files found')
    sys.exit(1)

# Test only first file for quick testing
files_list = files_list[:1]

# Test Loop
for k, filepath in enumerate(files_list, 1):
    print(f'[test_toolset_aout] [{k}/{len(files_list)}] [{filepath.name}]')

    aout_data = np.loadtxt(filepath, delimiter=',')
    dataset_name = filepath.stem
    sub_folder = output_dir / dataset_name / 'test_toolset_aout'

    status = toolset_aout(aout_data, sub_folder, visible=verbose, resolution=resolution)

    if status['success']:
        print('[All tools completed successfully]')
    else:
        print('[Some tools failed!]')
        for error in status['errors']:
            print(f'    - {error}')

print('\n=== Test complete ===')
